openpype/hosts/max/plugins/publish/validate_tyflow_data.py

```diff
@@ -0,0 +1,94 @@
+import pyblish.api
+from openpype.pipeline import PublishValidationError
+from pymxs import runtime as rt
+
+
+class ValidateTyFlowData(pyblish.api.InstancePlugin):
+    """Validate that TyFlow plugins or
+    relevant operators being set correctly."""
+
+    order = pyblish.api.ValidatorOrder
+    families = ["pointcloud", "tycache"]
+    hosts = ["max"]
+    label = "TyFlow Data"
+
+    def process(self, instance):
+        """
+        Notes:
+            1. Validate the container only include tyFlow objects
+            2. Validate if tyFlow operator Export Particle exists
+
+        """
+        report = []
+
+        invalid_object = self.get_tyflow_object(instance)
+        if invalid_object:
+            report.append(f"Non tyFlow object found: {invalid_object}")
+
+        invalid_operator = self.get_tyflow_operator(instance)
+        if invalid_operator:
+            report.append((
+                "tyFlow ExportParticle operator not "
+                f"found: {invalid_operator}"))
+        if report:
+            raise PublishValidationError(f"{report}")
```

Comment:
I would use raise PublishValidationError for invalid object and invalid operator respectively

Reply:
You are best of doing:

psuedocode:

```python
log.error("issue-1")
log.error("issue-2")
log.error("issue-3")
log.error("issue-4")
raise PublishValidationError("issues occurred", description="describe the validator and how the artist can solve any issues it might find")
```

By **logging directly** instead of building the report into a list first it also means that **if any unexpected error occurs** that the first few logs had been generated at least and it's much easier to pinpoint how far the code actually got. Plus the code gets more readable as well. :)